src/mavoscript.js

```diff
@@ -819,41 +795,13 @@ var _ = Mavo.Script = {
 		}
 	},
 
-	closest (node, type) {
-		let n = node;
-
-		do {
-			if (n.type === type) {
-				return n;
-			}
-		} while (n = n.parent);
-
-		return null;
-	},
+	closest: Vastly.closest,
 
 	serialize: (node, parent) => {
-		if (typeof node === "string") {
-			return node; // already serialized
-		}
-
 		if (parent) {
-			node.parent = parent;
-		}
-
-		var ret = _.transformations[node.type]?.(node, parent);
-
-		if (typeof ret == "object" && ret?.type) {
-			node = ret;
+			Vastly.parents.set(node, parent);
```

Comment:
I originally wanted to write this function as:
```js
serialize: (node) => {
    Vastly.parents.setAll(node);
    return Vastly.serialize(node);
},
```
However, due to `serializers` needing to pass a parent argument back into `serialize`, we have to do
```js
serialize: (node, parent) => {
    if (parent) {
        Vastly.parents.set(node, parent);
    }
    return Vastly.serialize(node);
},
```

Reply:
But is that parent argument used? It seems to be primarily used to set the parent, which we already do.

Reply:
That's what I thought too, but it fails the majority of the serialize tests when implented like that, I'm still working on trying to figure out why that's the case

Reply:
I will update once I figure it out

Reply:
Which tests does it fail?

Reply:
![Screen Shot 2023-12-05 at 5 27 48 PM](https://github.com/mavoweb/mavo/assets/45052692/e0949cfa-5203-4414-aae5-b5c3a34e851b)

It's all related to member expressions

Reply:
Hm, so it seems all related to it rewriting Mavo functions to use `get()` as well, so e.g. instead of `$fn.eq()` you get `$fn.get($fn, "eq")`, which happens [here](https://github.com/mavoweb/mavo/blob/main/src/mavoscript.js#L641). So you need to verify that what you're doing is actually setting parents, cause them not being set properly would be my top guess (or maybe they're not properly set after transformations).

Reply:
@LeaVerou  Coming back to this bug, it seems the root cause is that serializers mutate the node, meaning the parents originally set by vastly are no longer correct, an example is [here](https://github.com/mavoweb/mavo/blob/main/src/mavoscript.js#L617). 

I think having
```js
if (parent) {
        Vastly.parents.set(node, parent);
}
```
isn't the worst solution, but I'll try to think of a way around it

Reply:
Note that what a lot of this logic does is trying to find the topmost variable, which you already have a function for!
Then, it's rewriting to prepend with `$fn`, which seems useful enough in itself that might be worth including in vastly (in a more generic form). E.g. we'll definitely need this for rewriting data variables to branch off `$data` as well, and for rewriting `where` calls (to branch off a function call with the possible alternatives). If this is handled by vastly, parents can be adjusted by vastly and calling code doesn't have to care about maintaining parent pointers.

Reply:
I'll create an issue for it over in vastly

Reply:
https://github.com/mavoweb/vastly/issues/32